Python/pystate.c

```diff
@@ -1195,6 +1195,66 @@ _PyThread_CurrentFrames(void)
     return result;
 }
 
+/* The implementation of sys._current_exceptions(). This is intended to be
+   called with the GIL held, as it will be when called via
+   sys._current_frames(). It's possible it would work fine even without the GIL
```

Comment:
I have doubts that it would work fine without the GIL held because of: 1) auditing, 2) memory allocation.

Reply:
I copy/pasted the comment from `sys._current_frames` since they use the same mechanism. I can also factor the code and the comment elsewhere.

Reply:
You must not use the Python C API without holding the GIL: PyLong_FromUnsignedLong(), PyDict_SetItem(), etc. require to hold the GIL.